lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyDiagnosticParticipant.java

```diff
@@ -261,4 +300,214 @@ private void validateConfigElements(DOMDocument domDocument, List<Diagnostic> di
+                        domDocument);
+                String message = "ERROR: \"" + featureName + "\" versionless feature cannot be resolved. The versioned features do not have a platform in common.";
+                list.add(new Diagnostic(range, message, DiagnosticSeverity.Error, LIBERTY_LEMMINX_SOURCE, INCORRECT_FEATURE_CODE));
+            }
+            else if(commonPlatforms.size()>1) {
+                Range range = XMLPositionUtility.createRange(versionLessFeatureTextNode.getStart(), versionLessFeatureTextNode.getEnd(),
+                        domDocument);
+                String message = "ERROR: The \"" + featureName + "\" versionless feature cannot be resolved since there are more than one common platform. Specify a platform or a feature with a version to enable resolution.";
+                list.add(new Diagnostic(range, message, DiagnosticSeverity.Error, LIBERTY_LEMMINX_SOURCE, INCORRECT_FEATURE_CODE));
+            }
+            else{
+                checkForVersionlessPlatforms(domDocument, list, commonPlatforms, versionLessFeatureTextNode, featureName);
+            }
+        }
+        if (!preferredPlatforms.isEmpty()) {
+            checkForVersionlessPlatforms(domDocument, list, preferredPlatforms, versionLessFeatureTextNode, featureName);
+        }
+    }
+
+    /**
+     * check all platforms for versionless features
+     * @param domDocument
+     * @param list
+     * @param preferredPlatforms
+     * @param versionLessFeatureTextNode
+     * @param featureName
+     * @param libertyVersion
+     * @param libertyRuntime
+     * @param requestDelay
+     */
+    private void checkForVersionlessPlatforms(DOMDocument domDocument,
+                                              List<Diagnostic> list, Set<String> preferredPlatforms,
+                                              DOMNode versionLessFeatureTextNode, String featureName) {
+        LibertyRuntime runtimeInfo = LibertyUtils.getLibertyRuntimeInfo(domDocument);
+        String libertyVersion = runtimeInfo == null ? null : runtimeInfo.getRuntimeVersion();
+        String libertyRuntime = runtimeInfo == null ? null : runtimeInfo.getRuntimeType();
+        final int requestDelay = SettingsService.getInstance().getRequestDelay();
+        Set<String> allPlatforrmsForVersionLess = FeatureService.getInstance()
+                .getAllPlatformsForVersionLessFeature(featureName, libertyVersion, libertyRuntime, requestDelay, domDocument.getDocumentURI());
+        if (Sets.intersection(allPlatforrmsForVersionLess, preferredPlatforms).isEmpty() ){
+            Range range = XMLPositionUtility.createRange(versionLessFeatureTextNode.getStart(), versionLessFeatureTextNode.getEnd(),
+                    domDocument);
+            String message = "ERROR: The \"" + featureName + "\" versionless feature cannot be resolved. Specify a platform or a feature with a version to enable resolution.";
```

Comment:
This message is misleading if they did specify platforms. Maybe pass in a boolean indicating that the preferredPlatforms were indeed specified platform elements vs a common platform determined from versioned features. Then you can give a better message for the case where platforms were specified ==> 
"ERROR: The "+featureName+" versionless feature does not have a configured platform."

Reply:
Changed message

If platform is specified, message will be 
"ERROR: The "+featureName+" versionless feature does not have a configured platform."

If platform is not specified, message will be 
"ERROR: The " + featureName +  "versionless feature cannot be resolved. Specify a platform or a feature with a version to enable resolution."